### ISSUE TITLE
Location of remote SCP client is assumed

### ISSUE THREAD

**Issue by siggimoo** [bramvdbogaerde/go-scp#15]
The location of the remote SCP client is hardcoded to `/usr/bin/scp`. This should be adjustable since it will not necessarily be correct for all systems.

**Comment by bramvdbogaerde** [comment]
Indeed, good catch.

I will see if I can create an optional field or parameter for the location of the SCP binary on the remote.

**Comment by bramvdbogaerde** [comment]
I changed the way the SCP client structs are build.

I introduced a ClientConfigurer, which always more fine grained control over the SCP client client.

At the moment in contains the following configuration options:

* host (required)
* ssh.ClientConfig (required)
* timeout (optional, defaults to a minute)
* remoteBinary (optional, default to /usr/bin/scp)

A customized client can be created as follows:

```
// assuming clientConfig is defined elsewhere in your code
client := scp.NewConfigurer("example.net:22", &clientConfig).RemoteBinary("/my/custom/path/to/scp").Create()
```

I will close this issue for now, if you have any questions, or feel that this does not fully solve your problem, don't hestitate to open it again.
